eZ/Publish/Core/Repository/SiteAccessAware/SearchService.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * SearchService class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Repository\SiteAccessAware;
+
+use eZ\Publish\API\Repository\SearchService as SearchServiceInterface;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\Core\Repository\Helper\LanguageResolver;
+
+/**
+ * SiteAccess aware implementation of SearchService injecting languages where needed.
+ */
+class SearchService implements SearchServiceInterface
+{
+    /** @var \eZ\Publish\API\Repository\SearchService */
+    protected $service;
+
+    /** @var LanguageResolver */
+    protected $languageResolver;
+
+    /**
+     * Construct service object from aggregated service and LanguageResolver.
+     *
+     * @param \eZ\Publish\API\Repository\SearchService $service
+     * @param LanguageResolver $languageResolver
+     */
+    public function __construct(
+        SearchServiceInterface $service,
+        LanguageResolver $languageResolver
+    ) {
+        $this->service = $service;
+        $this->languageResolver = $languageResolver;
+    }
+
+    public function findContent(Query $query, array $languageFilter = array(), $filterOnUserPermissions = true)
+    {
+        $languageFilter['languages'] = $this->languageResolver->getPrioritizedLanguages(
+            empty($languageFilter['languages']) ? [] : $languageFilter['languages']
+        );
+
+        $languageFilter['useAlwaysAvailable'] = $this->languageResolver->getUseAlwaysAvailable(
+            empty($languageFilter['useAlwaysAvailable']) ? null : $languageFilter['useAlwaysAvailable']
+        );
+
+        return $this->service->findContent($query, $languageFilter, $filterOnUserPermissions);
+    }
+
+    public function findContentInfo(Query $query, array $languageFilter = array(), $filterOnUserPermissions = true)
+    {
+        $languageFilter['languages'] = $this->languageResolver->getPrioritizedLanguages(
+            empty($languageFilter['languages']) ? [] : $languageFilter['languages']
```

Comment:
it seems a little bit strange to check for empty and then pass empty array

Reply:
yes.. actually got really confused by this one myself now.

I think what I intended is default to load all languages like before for BC, however this kind of breaks the logic, and now that this layer is optional it does not make sense afaik.

So maybe we should rather do same as the other cases, if not set we intrepid that as null, and if set we pass whatever is there.

Sounds correct to you?

Reply:
I would rather always have `$languageFilter['languages']` available as empty array instead of null, but this seems good :)

Also, i just noticed there is old array syntax called in default argument, should be array `$languageFilter = []`